hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManager.java

```diff
@@ -65,7 +65,8 @@ public interface KeyManager extends OzoneManagerFS, IOzoneAcl {
    * @return a OmKeyInfo instance client uses to talk to container.
    * @throws IOException
    */
-  OmKeyInfo lookupKey(OmKeyArgs args, String clientAddress) throws IOException;
+  OmKeyInfo lookupKey(OmKeyArgs args, BucketLayout bucketLayout,
```

Comment:
Passing `BucketLayout` to `lookupKey`/`getKeyInfo` seems strange to me.  I would prefer passing `OmBucketInfo` (or `ResolvedBucket` if more convenient) instead:

1. to hide the implementation detail that we need `bucketLayout` for some internal logic
2. make future changes easier, should we need another bucket property

Reply:
That makes sense. Thanks. Changed to pass `ResolvedBucket`, `OmBucketInfo` is a cached thing so I'd avoid passing them around.

Reply:
Thanks @duongkame for updating `lookupKey` with this.  Have you intentionally left `getKeyInfo` with the `BucketLayout` parameter?

Reply:
Definitely not intentional, I missed it out of the blue.